### PR TITLE
feat: Transaction Processing logic & Wallet linking implementation

### LINKED ISSUE #6247: Implementation of PayTM Payment Gateway
**Feature Description & Workflow**
Integration of PayTM payment gateway is required to boost sales in India. The workflow would be in the following way

1. When an order is placed, we generate the checksumhash on the server-side.
2. to obtain the txn token, a call is to be made to the `Initiate Transaction API` with the checksum & parameters.
3. We fetch the available Payment Options from the `Fetch Payment Option API`.
4. User selects a payment option as Pay using PayTM
5. User enters mobile number linked to PayTM. Calls `Send OTP API` & `Validate OTP API`.
6. Call the `Transaction Status API` to verify the transaction response **[essential]**
7. Redirect the user to the appropriate order page in accordance with the response

### LINKED ISSUE #6353: Implement functions to process payTM transaction
**Is your feature request related to a problem? Please describe.**
Parent issue: #6247 
This is the second step in the workflow of PayTM transactions. After obtaining the txn token from the initiation process, we proceed to steps pertaining to fetching the payments options and processing the transaction.

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #6247 
Fixes #6353 
#### Short description of what this resolves:
This PR aims to fetch the payment options available for a particular order, validate one-time passwords and process the transactions.

#### Changes proposed in this pull request:

- Add URL helper methods to Payments Manager
- Added logic for processing transactions
- Send & validate one-time passwords - Route Logic

### CODE REVIEW

**Comment by mrsaicharan1** [comment]
@shreyanshdwivedi @uds5501 This is now ready for a review. Let's speed this up so that we can link it with the frontend pipeline.

**Comment by mrsaicharan1** [comment]
@shreyanshdwivedi @uds5501 I've made the requested changes. Please have a look now.

File: app/api/orders.py [hunk 1]
```diff
@@ -660,11 +660,105 @@ def initiate_transaction(order_identifier):
...
+            format(get_settings()['paytm_live_merchant'], order_identifier)
+
+    head = {
+        "clientId": "C11",
+        "version": "v1",
+        "requestTimestamp": str(int(time.time())),
+        "channelId": "WEB",
+        "txnToken": txn_token
+    }
+    body = {"mobileNumber": ""}
```
Comment (uds5501) [inline app/api/orders.py]: @mrsaicharan1 This needs to be provided in the data acquired by `POST`

File: app/api/orders.py [hunk 2]
```diff
@@ -660,11 +660,105 @@ def initiate_transaction(order_identifier):
...
+        url = "https://securegw.paytm.in/theia/api/v1/login/validateOtp?mid={}&orderId={}".\
+            format(get_settings()['paytm_live_merchant'], order_identifier)
+    head = {
+        "clientId": "C11",
+        "version": "v1",
+        "requestTimestamp": str(int(time.time())),
+        "channelId": "WEB",
+        "txnToken": txn_token
+    }
+    body = {"otp": ""}
```
Comment (uds5501) [inline app/api/orders.py]: @mrsaicharan1 Same goes for this. This needs to be acquired via post data.

File: app/api/orders.py [hunk 3]
```diff
@@ -660,11 +660,105 @@ def initiate_transaction(order_identifier):
...
+        "version": "v1",
+        "requestTimestamp": str(int(time.time())),
+        "channelId": "WEB",
+        "txnToken": txn_token
+    }
+    response = PaytmPaymentsManager.hit_paytm_endpoint(url=url, head=head)
+    return response
+
+
+@order_misc_routes.route('/orders/<string:order_identifier>/paytm/send_otp/<string:txn_token>', methods=['POST', 'GET'])
```
Comment (shreyanshdwivedi) [inline app/api/orders.py]: I think we should allow only POST in the methods

**Comment by mrsaicharan1** [comment]
@uds5501 @shreyanshdwivedi I've made the requested changes. This is ready for another review.

**Comment by mrsaicharan1** [comment]
I've also added the _JWT_ decorator to ensure proper routing based on authentication.

**Comment by kushthedude** [comment]
Shouldn't we have waited for @iamareebjamal review too?

**Pull request merged**